core/Tracker/Request.php

```diff
@@ -86,12 +86,25 @@ public function __construct($params, $tokenAuth = false)
             }
         }
 
-        // check for 4byte utf8 characters in url and replace them with �
+        // check for 4byte utf8 characters in all tracking params and replace them with �
         // @TODO Remove as soon as our database tables use utf8mb4 instead of utf8
-        if (array_key_exists('url', $this->params) && preg_match('/[\x{10000}-\x{10FFFF}]/u', $this->params['url'])) {
-            Common::printDebug("Unsupport character detected. Replacing with \xEF\xBF\xBD");
-            $this->params['url'] = preg_replace('/[\x{10000}-\x{10FFFF}]/u', "\xEF\xBF\xBD", $this->params['url']);
+        $this->params = $this->replaceUnsupportedUtf8Chars($this->params);
+    }
+
+    protected function replaceUnsupportedUtf8Chars($value, $key=false)
+    {
+        if (is_string($value) && preg_match('/[\x{10000}-\x{10FFFF}]/u', $value)) {
+            Common::printDebug("Unsupport character detected in $key. Replacing with \xEF\xBF\xBD");
+            return preg_replace('/[\x{10000}-\x{10FFFF}]/u', "\xEF\xBF\xBD", $value);
         }
+
+        if (is_array($value)) {
+            array_walk_recursive ($value, function(&$value, $key){
```

Comment:
as I haven't checked it... is it possible that there are nested arrays? reading on the documentation:
> You may notice that the key 'sweet' is never displayed. Any key that holds an array will not be passed to the function.

Also I presume it keeps indexes/keys?

Reply:
`array_walk_recursive` will "visit" all leafs of a nested array and it keeps indexes.